app/src/main/java/org/mozilla/fenix/settings/wallpaper/Extensions.kt

```diff
@@ -17,11 +17,18 @@ fun List<Wallpaper>.groupByDisplayableCollection(): Map<Wallpaper.Collection, Li
     val wallpapers = it.value.filter { wallpaper ->
         wallpaper.thumbnailFileState == Wallpaper.ImageFileState.Downloaded
     }
-    if (it.key.name == "classic-firefox") {
+    // Add the default wallpaper to the classic firefox collection
+    if (it.key.name == Wallpaper.classicFirefoxCollectionName) {
         it.key to listOf(Wallpaper.Default) + wallpapers
     } else {
         it.key to wallpapers
     }
-}.toMap().takeIf {
-    it.isNotEmpty()
-} ?: mapOf(Wallpaper.DefaultCollection to listOf(Wallpaper.Default))
+}.toMap().let { result ->
+    // Ensure the default is shown in the classic firefox collection even if those wallpapers are
+    // missing
+    if (result.containsKey(Wallpaper.ClassicFirefoxCollection)) {
```

Comment:
should this be the string and not the whole collection?
```suggestion
    if (result.containsKey(Wallpaper.classicFirefoxCollectionName)) {
```

Reply:
The keys are currently collections, but I will update the logic check just the name since we've run into problems there already. Good catch!